merlin-rxjava2/src/test/java/com/novoda/merlin/rxjava2/MerlinFlowableShould.java

```diff
@@ -0,0 +1,103 @@
+package com.novoda.merlin.rxjava2;
+
+import android.content.Context;
+
+import com.novoda.merlin.Merlin;
+import com.novoda.merlin.NetworkStatus;
+import com.novoda.merlin.registerable.bind.Bindable;
+import com.novoda.merlin.registerable.connection.Connectable;
+import com.novoda.merlin.registerable.disconnection.Disconnectable;
+
+import io.reactivex.disposables.Disposable;
+import io.reactivex.subscribers.TestSubscriber;
+
+import java.util.List;
+
+import org.junit.Test;
+
+import org.mockito.ArgumentCaptor;
+
+import static com.novoda.merlin.rxjava2.MerlinFlowable.from;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+
+public class MerlinFlowableShould {
+
+    @Test
+    public void unbindWhenDisposed() {
+        Merlin merlin = mock(Merlin.class);
+        Disposable disposable = from(merlin).subscribe();
+        disposable.dispose();
+
+        verify(merlin, times(1)).unbind();
+    }
+
+    @Test
+    public void receiveOneAvailableNetworkStatusStateOnConnect() {
+        Merlin merlin = mock(Merlin.class);
+        ArgumentCaptor<Connectable> argumentCaptor = ArgumentCaptor.forClass(Connectable.class);
+        TestSubscriber<NetworkStatus.State> testSubscriber = MerlinFlowable.from(merlin)
+                                                                           .test();
+
+        verify(merlin).registerConnectable(argumentCaptor.capture());
+        argumentCaptor.getValue()
+                      .onConnect();
+        List<NetworkStatus.State> stateList = testSubscriber.values();
```

Comment:
Same as the `MerlinObservableShould` class, we can pull out fields and utility methods passing a `NetworkStatus.State`.